src/v/datalake/schemaless_translator.cc

```diff
@@ -26,7 +26,7 @@ iceberg::struct_value schemaless_translator::translate_event(
     return res;
 }
 
-iceberg::struct_type schemaless_translator::get_schema() const {
+iceberg::struct_type schemaless_translator::schema() {
```

Comment:
nit: maybe we can move this to separate file something like `table_definition.h` ?

Reply:
Done